### PR TITLE
[FIX] website: fix Go to website and website selector

### PR DESCRIPTION
**Pull request by rdeodoo**
rebase-ff

### CODE REVIEW

**Comment by rdeodoo** [comment]
@qsm-odoo About `website_root` I dit not have to share code as it was simpler in dashboard (don't need to reload on same url).

I took the opportunity to fix it since it was not working if `http://` or `:8069` was added in website domain (but it is supported for other flows).

Most of the code was remaining of the first implementation that forced the website in session before redirecting anyway (it was changed to use `fw=`).

File: addons/website/controllers/backend.py [hunk 1]
```diff
@@ -26,8 +26,10 @@ def fetch_dashboard_data(self, website_id, date_from, date_to):
 
         current_website = website_id and Website.browse(website_id) or Website.get_current_website()
         multi_website = request.env.user.has_group('website.group_multi_website')
-        dashboard_data['websites'] = (multi_website and request.env['website'].search([]) or current_website).read(['id', 'name'])
-        for website in dashboard_data['websites']:
+        websites = multi_website and request.env['website'].search([]) or current_website
+        dashboard_data['websites'] = websites.read(['id', 'name', 'domain'])
+        for rec, website in zip(websites, dashboard_data['websites']):
```
Comment (rdeodoo) [inline addons/website/controllers/backend.py]: @JKE-be Indeed `zip()` better than tweaking `_get_http_domain()` to work with a string (and not only a record)

File: addons/website/controllers/backend.py [hunk 2]
```diff
@@ -26,8 +26,10 @@ def fetch_dashboard_data(self, website_id, date_from, date_to):
 
         current_website = website_id and Website.browse(website_id) or Website.get_current_website()
         multi_website = request.env.user.has_group('website.group_multi_website')
-        dashboard_data['websites'] = (multi_website and request.env['website'].search([]) or current_website).read(['id', 'name'])
-        for website in dashboard_data['websites']:
+        websites = multi_website and request.env['website'].search([]) or current_website
+        dashboard_data['websites'] = websites.read(['id', 'name', 'domain'])
```
Comment (qsm-odoo) [inline addons/website/controllers/backend.py]: remove domain :)

File: addons/website/static/src/js/backend/dashboard.js [hunk 3]
```diff
@@ -320,10 +326,16 @@ var Dashboard = AbstractAction.extend(ControlPanelMixin, {
                 self.on_website_button($(ev.target).data('website-id'));
             });
         }
+
+        var $buttons = $(QWeb.render("website.GoToButtons"));
+        $buttons.find('a.js_go_to_website').addBack('a.js_go_to_website').click(function (ev) {
```
Comment (qsm-odoo) [inline addons/website/static/src/js/backend/dashboard.js]: - Just $buttons.click(
- `.on('click', this.on_go_to_website.bind(this))`

File: addons/website/static/src/xml/website.backend.xml [hunk 4]
```diff
@@ -103,7 +103,7 @@
     </t>
 
     <t t-name="website.GoToButtons">
-        <a role="button" href="/" class="btn btn-primary" title="Go to Website">
+        <a role="button" href="#" class="btn btn-primary js_go_to_website" title="Go to Website">
```
Comment (qsm-odoo) [inline addons/website/static/src/xml/website.backend.xml]: No need of the new class.

File: addons/website/controllers/backend.py (hunk 2 above)
Comment (rdeodoo) [reply addons/website/controllers/backend.py]: Indeed, leftover of previous version without `zip()`, done

File: addons/website/static/src/js/backend/dashboard.js (hunk 3 above)
Comment (rdeodoo) [reply addons/website/static/src/js/backend/dashboard.js]: Done, as discussed was to ensure targeting the button if the DOM change someday but should not think that far

**Comment by rdeodoo** [comment]
Processed & Rebuilt

File: addons/website/static/src/xml/website.backend.xml [hunk 5]
```diff
@@ -103,7 +103,7 @@
     </t>
 
     <t t-name="website.GoToButtons">
-        <a role="button" href="/" class="btn btn-primary" title="Go to Website">
```
Comment (qsm-odoo) [inline addons/website/static/src/xml/website.backend.xml]: Don't change that (middle click)

Comment (rdeodoo) [reply addons/website/static/src/xml/website.backend.xml]: Done, sadly it [won't work with ctrl+click](https://superuser.com/questions/854797/why-does-ctrl-click-not-open-some-links-in-a-new-tab) but wheel click will, that's better than nothing

**Comment by qsm-odoo** [comment]
@robodoo delegate+

**Comment by rdeodoo** [comment]
@robodoo r+

**Pull request closed without merging**